src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4153,16 +4155,43 @@ public override bool PreVisit(CallNode node)
                     return false;
                 }
 
+                var numOverloads = overloads.Count();
+
+                var overloadsWithUntypedObjectLambdas = overloadsWithLambdas.Where(func => func.ParamTypes.Any() && func.ParamTypes[0] == DType.UntypedObject);
+                TexlFunction overloadWithUntypedObjectLambda = null;
+                if (overloadsWithUntypedObjectLambdas.Any())
+                {
+                    Contracts.Assert(overloadsWithUntypedObjectLambdas.Count() == 1, "Incorrect multiple overloads with both UntypedObject and lambdas.");
+                    overloadWithUntypedObjectLambda = overloadsWithUntypedObjectLambdas.Single();
+                    Contracts.Assert(overloadWithUntypedObjectLambda.HasLambdas);
```

Comment:
>Contracts.Assert(overloadWithUntypedObjectLambda.HasLambdas); [](http://example.com/codeflow?start=20&length=61)

This assert is not needed as this is already a subset of overloadsWithLambdas.

Reply:
Fixed